org.palladiosimulator.analyzer.slingshot.behavior.spd/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/interpreter/entity/constraint/ThrashingConstraintFilter.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * 
+ */
+package org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entity.constraint;
+
+import org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entities.FilterObjectWrapper;
+import org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entities.FilterResult;
+import org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entities.TargetGroupState;
+import org.palladiosimulator.spd.ScalingPolicy;
+import org.palladiosimulator.spd.adjustments.AbsoluteAdjustment;
+import org.palladiosimulator.spd.adjustments.AdjustmentType;
+import org.palladiosimulator.spd.adjustments.RelativeAdjustment;
+import org.palladiosimulator.spd.adjustments.StepAdjustment;
+import org.palladiosimulator.spd.constraints.target.ThrashingConstraint;
+
+/**
+ * The thrashing constraint filter makes sure that there is a certain amount of
+ * time passed since the last enacted policy that adjusted the amount of
+ * resources in the opposite direction in comparison to the one being enacted.
+ * 
+ * @author Floriment Klinaku
+ *
+ */
+public class ThrashingConstraintFilter extends AbstractConstraintFilter<ThrashingConstraint> {
+
+	private enum ADJUSTMENT_SIGN {
+		POSITIVE,
+		NEGATIVE
+	}
+	
+	
+	public ThrashingConstraintFilter(ThrashingConstraint constraint) {
+		super(constraint);
+		// TODO Auto-generated constructor stub
+	}
+
+	@Override
+	public FilterResult doProcess(FilterObjectWrapper event) {
+			
+		TargetGroupState targetGroupState = event.getState().getTargetGroupState();
+		
+		if(targetGroupState.enactedPoliciesEmpty()) {
+			//no previously executed policies, thrashing passes
+			return FilterResult.success(event.getEventToFilter());
+		}
+		
+		ScalingPolicy lastEnactedPolicy = targetGroupState.getLastEnactedScalingPolicy();
+		double lastSimulationTime = targetGroupState.getLastScalingPolicyEnactmentTime();
+		double currentSimulationTime = event.getEventToFilter().time();
+		
+		ScalingPolicy currentScalingPolicy = event.getState().getScalingPolicy();
+		
+		if(currentScalingPolicy.getAdjustmentType() instanceof AbsoluteAdjustment
```

Comment:
this check should be different in my opinion. 
e.g. you do an scale out with an absolute adjustment, and right after you have an absolute adjustment to a smaller goal value. in my opinion this should be considered a case of thrashing, but this check would pass it as "success".

I see two posibillities:
1. adapt the SPD documentation, and explicitly exclude Absolute Adjustments from the thrashing constraint. (thereby altering my expectation)
2. somehow determine whether an absolute adjustemtn was positive, or negative. Maybe by adding more information to the `targetGroupState`?